src/G2/Execution/Rules.hs

```diff
@@ -133,8 +137,67 @@ mapAccumMaybeM f s xs = do
 
 evalVarSharing :: State t -> NameGen -> Id -> (Rule, [State t], NameGen)
 evalVarSharing s@(State { expr_env = eenv
-                        , exec_stack = stck })
+                        , exec_stack = stck
+                        , poly_arg_map = pargm
+                        , ty_app_re_map = tarm })
                ng i
+    -- The value being evaluated is a symbolic type variable, which will only occur as
```

Comment:
Symbolic _value_ variable with a type that is a type variable.  We can sometimes have these in cases besides polymorphic functions also.